### PR TITLE
Matgl upgrade to 1.1.3 + CHGNet Integration

### PR DESCRIPTION
**Pull request by melo-gonzo**
This PR takes over dependabot PR #269 and also includes the addition of CHGNet from `matgl` which was introduced in `matgl-v1.1.0`. The PR also includes updates to `TensorNet` forward pass which still must be copied in manually.

### CODE REVIEW

**Review by laserkelvin — COMMENTED** [review]
For the most part it looks good to me, but I wasn't 100% sure what you were referring to RE: the TensorNet forward copying changes?

File: matsciml/models/dgl/chgnet.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+        Args:
+            g (dgl.DGLGraph): Input g.
+            state_attr (torch.Tensor, optional): State features. Defaults to None.
+            l_g (dgl.DGLGraph, optional): Line graph. Defaults to None and is computed internally.
+
+        Returns:
+            torch.Tensor: Model output.
+        """
+        # g.edata["pbc_offshift"] = g.edata["offsets"]
```
[the remark this answers is not in the document]
Comment (melo-gonzo) [reply matsciml/models/dgl/chgnet.py]: purged em [991edef](https://github.com/IntelLabs/matsciml/pull/273/commits/991edef2412211d3aa5ee3b6a50da05befb5e884)

File: experiments/configs/models/chgnet_dgl.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+encoder_class:
+  class_path: matsciml.models.M3GNet
```
[the remark this answers is not in the document]
Comment (melo-gonzo) [reply experiments/configs/models/chgnet_dgl.yaml]: fixed this [d436989](https://github.com/IntelLabs/matsciml/pull/273/commits/d4369895175af9fa2d3b0a09b32a1fa39d984464)

**Comment by melo-gonzo** [comment]
> For the most part it looks good to me, but I wasn't 100% sure what you were referring to RE: the TensorNet forward copying changes?

Just that we have to copy-paste in the forward method for TensorNet (and CHGNet) since they dont offer the same interface as the M3GNet model where we can return the embeddings only with `return_all_layer_output=True`.

**Review by laserkelvin — APPROVED** [review]

**Comment by laserkelvin** [comment]
Please go ahead and merge!

**Pull request merged**